content/_code-samples/key-derivation/js/index.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+
+//organize imports
+const assert = require("assert")
+const brorand = require("brorand")
+const BN = require("bn.js")
+const elliptic = require("elliptic");
+const Ed25519 = elliptic.eddsa('ed25519');
+const Secp256k1 = elliptic.ec('secp256k1');
+const { rfc1751MnemonicToKey, keyToRFC1751Mnemonic } = require("xrpl/dist/npm/Wallet/rfc1751");
+const hashjs = require("hash.js");
+const Sha512 = require("ripple-keypairs/dist/Sha512")
+const { codec, encodeAccountPublic, encodeNodePublic } = require("ripple-address-codec");
+
+const XRPL_SEED_PREFIX = 0x21
+
+const isHex = function(value) {
+    const regex = new RegExp(/^0x([0-9a-f]+(\.[0-9a-f]*)?|\.[0-9a-f]+)(p[+-]?\d+)?$/, 'i')
```

Comment:
Why does this regex have `\.` in it? 
Do you have examples of inputs which would trigger the logic beyond the first `/^0x[0-9a-f]+`?

Also, I think this might not hit upper case hex, which should be considered hex

Reply:
The `,'i'` at the end makes this RegExp case-insensitive, so that part is OK.

I'm not sure what `(\.[0-9a-f]*)?|\.[0-9a-f]+)(p[+-]?\d+)??` is about though. This regex accepts `0x1234.1` as an example value, which is not really correct.

For what it's worth, in XRP Ledger code, we usually don't use a prefix like `0x` for hex strings either.

It's a drop-in replacement given how this function is used elsewhere, but I would prefer this code use a simpler regex like the following:

```suggestion
    const regex = new RegExp(/^[0-9a-f]+$/, 'i')
```

(You could also constrain the length, for example by replacing the `+` with `{2,64}` which results in 1-32 bytes when decoded since two hex characters makes 1 byte.)